compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -3916,23 +3927,53 @@ private BLangMatchPattern transformMatchPattern(Node matchPattern, Location matc
                 bLangListMatchPattern.setRestMatchPattern(
-                        (BLangRestMatchPattern) transformMatchPattern(restMatchPatternNode,
-                                getPosition(restMatchPatternNode)));
+                        (BLangRestMatchPattern) transformMatchPattern(restMatchPatternNode));
             }
             return bLangListMatchPattern;
-        } else if (matchPattern.kind() == SyntaxKind.REST_MATCH_PATTERN) {
+        }
+
+        if (kind == SyntaxKind.REST_MATCH_PATTERN) {
             RestMatchPatternNode restMatchPatternNode = (RestMatchPatternNode) matchPattern;
             BLangRestMatchPattern bLangRestMatchPattern = (BLangRestMatchPattern) TreeBuilder.createRestMatchPattern();
             bLangRestMatchPattern.pos = matchPatternPos;
 
             SimpleNameReferenceNode variableName = restMatchPatternNode.variableName();
             bLangRestMatchPattern.setIdentifier(createIdentifier(getPosition(variableName), variableName.name()));
             return bLangRestMatchPattern;
-        } else {
-            // TODO : Remove this after all binding patterns are implemented
-            dlog.error(matchPatternPos, DiagnosticCode.MATCH_PATTERN_NOT_SUPPORTED);
-            return null;
         }
+
+        if (kind == SyntaxKind.MAPPING_MATCH_PATTERN) {
+            MappingMatchPatternNode mappingMatchPatternNode = (MappingMatchPatternNode) matchPattern;
+            BLangMappingMatchPattern bLangMappingMatchPattern =
+                    (BLangMappingMatchPattern) TreeBuilder.createMappingMatchPattern();
+            bLangMappingMatchPattern.pos = matchPatternPos;
+
+            for (FieldMatchPatternNode fieldMatchPatternNode : mappingMatchPatternNode.fieldMatchPatterns()) {
+                bLangMappingMatchPattern.fieldMatchPatterns.add((BLangFieldMatchPattern)
+                        transformMatchPattern(fieldMatchPatternNode));
+            }
+            if (mappingMatchPatternNode.restMatchPattern().isPresent()) {
+                bLangMappingMatchPattern.restMatchPattern =
+                        (BLangRestMatchPattern) transformMatchPattern(mappingMatchPatternNode.restMatchPattern().get());
+            }
+
+            return bLangMappingMatchPattern;
+        }
+
+        if (kind == SyntaxKind.FIELD_MATCH_PATTERN) {
+            FieldMatchPatternNode fieldMatchPatternNode = (FieldMatchPatternNode) matchPattern;
+            BLangFieldMatchPattern bLangFieldMatchPattern =
+                    (BLangFieldMatchPattern) TreeBuilder.createFieldMatchPattern();
+
+            bLangFieldMatchPattern.fieldName =
+                    createIdentifier(fieldMatchPatternNode.fieldNameNode());
+            bLangFieldMatchPattern.matchPattern = transformMatchPattern(fieldMatchPatternNode.matchPattern());
+            return bLangFieldMatchPattern;
+        }
+
+        // TODO : Remove this after all binding patterns are implemented
```

Comment:
Do we have an issue to track cleaning these up?

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/24607